### PR TITLE
Minor edits to benchmarking code

### PR DESCRIPTION
**Pull request by quaquel**
* fix for < python 3.12
* Insert the current repo of mesa into sys.path to ensure you are using the correct version
* pep8 name changes of files
* added code to run each benchmark in isolation (useful for profiling and line profiling, used it extensively already for https://github.com/projectmesa/mesa/issues/1900#issuecomment-1902643678)

### CODE REVIEW

**Comment by EwoutH** [comment]
Let's see if the CI is working: /rerun-benchmarks

File: .gitignore [hunk 1]
```diff
@@ -88,3 +88,5 @@ pythonenv*/
 # JS dependencies
 mesa/visualization/templates/external/
 mesa/visualization/templates/js/external/
+*.pclprof
+*.pickle
```
Comment (EwoutH) [inline .gitignore:92]: On line 2 `benchmarking/**/*.pickle` is already defined. Is it needed to ignore pickles everywhere?

**Comment by EwoutH** [comment]
Let me know when this is ready for review

File: .gitignore (hunk 1 above)
Comment (quaquel) [reply .gitignore:92]: I thought, too, that you added them to .gitignore, but somehow it was not picked up on my machine. I will take a closer look.

I also have a few more minor changes coming due to the cell space comparison (see also https://github.com/projectmesa/mesa/issues/1900#issuecomment-1902643678)

Comment (EwoutH) [reply .gitignore:92]: Maybe rebase your branch on main, it looks like it's lagging a bit behind.

**Comment by quaquel** [comment]
> Let me know when this is ready for review

It should be alright now. Please squash and merge this because I made a bit of a mess of the commit history.

**Comment by EwoutH** [comment]
Could you update the PR message with a bit of motivation for the changes made?

/rerun-benchmarks

**Comment by quaquel** [comment]
> Could you update the PR message with a bit of motivation for the changes made?
> 
> /rerun-benchmarks

I already did so, or do you want to know more? But if so, what?

File: benchmarks/WolfSheep/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from wolf_sheep import WolfSheep
```
Comment (quaquel) [inline benchmarks/WolfSheep/__init__.py:1]:
```suggestion
from wolf_sheep import WolfSheep
```

File: .gitignore (hunk 1 above)
Comment (EwoutH) [reply .gitignore]: It should indeed have been `benchmarks/**/*.pickle`, good catch

**Comment by EwoutH** [comment]
LGTM, thanks for this effort. @rht or @Corvince could one of you do the final review and squash merge if ok?

Did you also observe the Schelling being faster on your system? Or is that an anomaly?

In the long run we should try to keep these models in sync with [ABMFrameworksComparison](https://github.com/JuliaDynamics/ABMFrameworksComparison). Do we want to make further improvements on these models in the short run?

**Comment by rht** [comment]
> In the long run we should try to keep these models in sync with [ABMFrameworksComparison](https://github.com/JuliaDynamics/ABMFrameworksComparison). Do we want to make further improvements on these models in the short run?

The sync with Agents.jl devs needs to happen soon, if possible. I think they would appreciate to know if we are now monitoring the performance as well.

**Comment by rht** [comment]
> Did you also observe the Schelling being faster on your system? Or is that an anomaly?

Haven't tested, but the finding is probably not a blocker to this PR being merged.

**Comment by quaquel** [comment]
> LGTM, thanks for this effort. @rht or @Corvince could one of you do the final review and squash merge if ok?
> 
> Did you also observe the Schelling being faster on your system? Or is that an anomaly?
> 
> In the long run we should try to keep these models in sync with [ABMFrameworksComparison](https://github.com/JuliaDynamics/ABMFrameworksComparison). Do we want to make further improvements on these models in the short run?

1. I think there is still quite some noise in the benchmarks. But one issue with the old schelling code was that it did not use self.random properly in all places. This probably contributes to even more noise.
2. In my view we should indeed keep our versions in sync with [ABMFrameworksComparison](https://github.com/JuliaDynamics/ABMFrameworksComparison). It would be good to push any changes to them in the context of a new stable release of MESA.

I also noticed that, in particular, WolfSheep is not a good Python implementation. There is a lot of unnecessary code duplication going on. I have the impression that some implementations were done to mimic NetLogo's version. Which is fine for a tutorial for people coming from NetLogo. However, I would argue that the benchmark (and most examples) should reflect the Pythonic and Mesa way of doing things. I am happy to push a WolfSheep reimplementation if desired (But also see [https://github.com/quaquel/mesa/blob/cell-space/benchmarks/WolfSheep/wolf_sheep.py](https://github.com/quaquel/mesa/blob/cell-space/benchmarks/WolfSheep/wolf_sheep.py), for the direction in which I would change it.

**Comment by EwoutH** [comment]
Sounds good, would you like to do both (pushing to [ABMFrameworksComparison](https://github.com/JuliaDynamics/ABMFrameworksComparison) and speeding up WolfSheep yourself?

**Comment by quaquel** [comment]
Yes, I'm fine with doing both (although it is more about cleaning up wolfsheep code; I have no idea about performance, but it is probably small).

**Comment by rht** [comment]
I think the first step is to simplify the WolfSheep at https://github.com/projectmesa/mesa-examples, then next all the other WolfSheep's (in Mesa's benchmark and Agents.jl's benchmark) should be updated to be almost the same as this reference implementation.

**Pull request merged**